lib/graphql/subscriptions/serialize.rb

```diff
@@ -123,6 +123,17 @@ def dump_value(obj)
             { SYMBOL_KEY => obj.to_s }
           elsif obj.respond_to?(:to_gid_param)
             {GLOBALID_KEY => obj.to_gid_param}
+          elsif defined?(ActiveSupport::TimeWithZone) && obj.is_a?(ActiveSupport::TimeWithZone)
+            # This handles a case where Rails prior to 7 would
+            # make the class ActiveSupport::TimeWithZone return "Time" for
+            # its name. In Rails 7, it will now return "ActiveSupport::TimeWithZone",
+            # which happens to be incompatible with expectations we have
+            # with what a Time class supports ( notably, strptime in `load_value` ).
+            #
+            # This reverts to what was happening before, in that ActiveSupport::TimeWithZone
+            # was coerced to a Time on the deserialization side.
+            #
+            { TIMESTAMP_KEY => [Time.name, obj.strftime(TIMESTAMP_FORMAT)] }
```

Comment:
Will this cause `TimeWithZone`s to be serialized identical to `Time`s, then when they're reloaded, they'll be loaded as `Time`s instead of `TimeWithZone`s? 

I guess that's what _used_ to happen, but why not instead, improve it to serialize and load in such a way to reload `TimeWithZone`s?

Reply:
Sure, I made that change - it requires adding awareness on the deserialize side of the incoming zone, therefore, the format of the arguments when passing a `TimeWithZone` must be changed.